protocol/app/process/other_msgs.go

```diff
@@ -59,8 +59,10 @@ func DecodeOtherMsgsTx(decoder sdk.TxDecoder, txBytes []byte) (*OtherMsgsTx, err
 // Validate returns an error if one of the underlying msgs fails `ValidateBasic`.
 func (omt *OtherMsgsTx) Validate() error {
 	for _, msg := range omt.msgs {
-		if err := msg.ValidateBasic(); err != nil {
-			return getValidateBasicError(msg, err)
+		if m, ok := msg.(sdk.HasValidateBasic); ok {
```

Comment:
When do we NOT expect `HasValidateBasic` to be the case? 

Should we handle else case?

Reply:
Many of the Cosmos SDK messages have migrated away from using `ValidateBasic` which I filed https://github.com/cosmos/cosmos-sdk/issues/18750 about. Cosmos SDK wants users to simulate their transactions to validate them instead of relying on `ValidateBasic` during `CheckTx` which is a change in expected user behavior going forward.

This is one of the points I brought up that we want to discuss in slack since this has ramifications to user experience.

Reply:
Blocking comment: what's the conclusion of this discussion?

Reply:
The conclusion was to simulate the transaction.

Reply:
Gotcha, nit: would you be able to add a comment explaining this?